pkg/build/cleanup.go

```diff
@@ -11,8 +11,16 @@ import (
 // temporary directories created by STI build and it also cleans the temporary
 // Docker images produced by LayeredBuild
 type DefaultCleaner struct {
-	util.FileSystem
-	docker.Docker
+	fs     util.FileSystem
+	docker docker.Docker
```

Comment:
> effectively make the parent struct be able to serve as (be passed as) the child struct

@bparees this is not true, because the types are different!
What may come close to that is transitively implementing interfaces.

If `T` is embed in `S` and `T` implements `I`, then `S` implements `I` too.